targeted/json_csrf_poc_generator

```diff
@@ -1,38 +1,43 @@
 //csrf poc generater supporting json csrf
-//it will print results to the zap script console. you will have to copy paste it from there
+//it will copy the results to clipboard and print them to the zap script console
 // released under the Apache v2.0 licence.
 //You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 
 function invokeWith(msg) {
-	print("\n\n\n");
-	print("<!DOCTYPE html>\n");
-	print("<head>\n <title>CSRF POC</title>\n</head>");
-	print("<body>");
+	string = "<!DOCTYPE html>\n";
+	string += "<head>\n <title>CSRF POC</title>\n</head>";
+	string += "\n<body>";
 	if(msg.getRequestHeader().getMethod().toString()== "POST"){
 	body = msg.getRequestBody().toString();
 	body = body.trim();
 	if(isJson(body))
-		print("<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\" enctype=\"text/plain\">" );
+		string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\" enctype=\"text/plain\">" ;
 	else
-		print("<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\">");
+		string += "\n<form action=\""+ msg.getRequestHeader().getURI().toString() +"\" id=\"formid\" method=\"post\">";
 	if(!isJson(body)){	
 	body=body.split('&');
 	for(i=0;i<body.length;i++)
 	{
 		keyval = body[i].split('=')
-		print("<input type=\"hidden\" name=\""+decodeURIComponent(keyval[0])+"\" value=\""+decodeURIComponent(keyval[1])+"\" />");	
+		string += "\n<input type=\"hidden\" name=\""+decodeURIComponent(keyval[0])+"\" value=\""+decodeURIComponent(keyval[1])+"\" />";	
```

Comment:
Whatever the situation is, I don't think that's needed for a csrf poc
generator.
Considering what a normal user needs.

On Thu, Aug 13, 2015, 2:28 PM thc202 <notifications@github.com> wrote:

> In targeted/json_csrf_poc_generator
> <https://github.com/zaproxy/community-scripts/pull/10#discussion_r36953166>
> :
>
> >  	if(!isJson(body)){	
> >  	body=body.split('&');
> >  	for(i=0;i<body.length;i++)
> >  	{
> >  		keyval = body[i].split('=')
> > -		print("<input type=\"hidden\" name=\""+decodeURIComponent(keyval[0])+"\" value=\""+decodeURIComponent(keyval[1])+"\" />");	
> > +		string += "\n<input type=\"hidden\" name=\""+decodeURIComponent(keyval[0])+"\" value=\""+decodeURIComponent(keyval[1])+"\" />";	
>
> True, that's just an extreme example, but other "valid" (i.e. more normal)
> values can break the HTML too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zaproxy/community-scripts/pull/10/files#r36953166>.
>